packages/kit/src/core/prerender/prerender.js

```diff
@@ -14,25 +14,17 @@ import { escape_html_attr } from '../../utils/escape.js';
  * @typedef {import('types').Logger} Logger
  */
 
-/** @type {(details: Parameters<PrerenderErrorHandler>[0] ) => string} */
-function format_error({ status, path, referrer, referenceType }) {
-	return `${status} ${path}${referrer ? ` (${referenceType} from ${referrer})` : ''}`;
-}
+/**
+ * @param {Parameters<PrerenderErrorHandler>[0]} details
+ * @param {import('types').ValidatedKitConfig} config
+ */
+function format_error({ status, path, referrer, referenceType }, config) {
+	const message =
+		status === 404 && !path.startsWith(config.paths.base)
+			? `${path} does not begin with \`base\`, which can be imported from \`$app/paths\``
+			: path;
 
-/** @type {(log: Logger, onError: OnError) => PrerenderErrorHandler} */
-function normalise_error_handler(log, onError) {
```

Comment:
Personal preference: I liked it better when this was a separate function (even if it's only used in one place). What was your reasoning for inlining it?